File: Assets/Levels/Plateformer/moveLeftAction.js
#pragma strict

public class moveLeftAction extends AbstractAction {

	public function moveLeftAction(pPlayer : GameObject){
		super(pPlayer);
	}

	public function execute(duration : int) : boolean{
		while(curTime < duration){
			player.transform.position.x -= 0.05;
			curTime++;
			return false;
		}
		curTime = 0;
		return true;
		
	}
}

File: Assets/Levels/Plateformer/OldMovementSystem/moveLeftAction.cs
using UnityEngine;
using System.Collections;

public class moveLeftAction : AbstractAction {

	public moveLeftAction(GameObject pPlayer) : base(pPlayer){

	}
	
	public override bool execute(int duration) {
		if(curTime < duration){
			player.transform.position -= new Vector3(0.05f, 0, 0);
			curTime++;
			return false;
		}
		else{
			curTime = 0;
			return true;
		}
	}

}